### ISSUE TITLE
Linear Pauli Rotations Documentation

### ISSUE THREAD

**Issue by MattiaIppoliti** [Qiskit/qiskit-terra#9861]
For a register of state qubits $\ket{x}$, a target qubit $\ket{0}$ and the basis 'Y' this circuit acts as, the operation $Y$ of a linear  rotation on a zero-input qubit is:

$$\left|x\right\rangle _{n}\left|0\right\rangle
\rightarrow\left|x\right\rangle _{n}\left(
\cos\left(\frac{a}{2}x+\frac{b}{2}\right)\left|0\right\rangle
+\sin\left(\frac{a}{2}x+\frac{b}{2}\right)\left|1\right\rangle \right)$$

Such a rotation can be realized as a series of controlled rotations as follows:

$$\left[R_{y}\left(2^{n-1}a\right)\right]^{x_{n-1}}\cdots
\left[R_{y}\left(2^{1}a\right)\right]^{x_{1}}
\left[R_{y}\left(2^{0}a\right)\right]^{x_{0}}R_{y}\left(b\right)$$

However, given the documentation: https://qiskit.org/documentation/stubs/qiskit.circuit.library.LinearPauliRotations.html , we find: 

$$|x\rangle |0\rangle \mapsto \cos(ax + b)|x\rangle|0\rangle + \sin(ax + b)|x\rangle |1\rangle$$

with $\frac{a}{2}=$ `slope` and $\frac{b}{2}=$ `offset`. I guess you should replace `a` and `b` respectively with `slope` and `offset` inside the last formula or use the first formula written above.

This is also why within the code at link: https://github.com/Qiskit/qiskit-terra/blob/3763e61f16f97598cb0fa317277ad85fa15db738/test/python/circuit/library/test_functional_pauli_rotations.py, we can see how correctly multiplied by a factor of two both a and b:

```py
def test_linear_function(self, num_state_qubits, slope, offset):
        """Test the linear rotation arithmetic circuit."""

        def linear(x):
            return offset + slope * x

        linear_rotation = LinearPauliRotations(num_state_qubits, slope * 2, offset * 2)
        self.assertFunctionIsCorrect(linear_rotation, linear)

```

- Link documentation: https://qiskit.org/documentation/stubs/qiskit.circuit.library.LinearPauliRotations.html

**Comment by 1ucian0** [comment]
@Cryoris, do you mind confirming the mistake? The math is over my head.

**Comment by jakelishman** [comment]
Thanks for spotting this.

I agree there's a problem here, though I think you might have got your suggestion slightly muddled too (or I misunderstood what you're trying to say with the replacements).  I think our documentation is off by a factor of 4, whereas your suggestions are off by a factor of 2 with respect to the `slope` and `offset` arguments.

The  output from `LinearPauliRotations(slope, offset)` in the $Y$ basis is definitely $\lvert x\rangle\lvert 0\rangle \to \cos\Bigl(\frac{\text{slope}}{2}x + \frac{\text{offset}}{2}\Bigr)\lvert x\rangle\lvert 0\rangle + \sin\Bigl(\frac{\text{slope}}{2}x + \frac{\text{offset}}{2}\Bigr)\lvert x\rangle\lvert 1\rangle$ (since it's implemented the same way as your first formula):
```python
In [1]: import math
   ...: import numpy as np
   ...: from qiskit import QuantumCircuit, QuantumRegister
   ...: from qiskit.circuit.library import LinearPauliRotations
   ...: from qiskit.quantum_info import Statevector
   ...:
   ...: target = QuantumRegister(1)
   ...: control = QuantumRegister(5)
   ...: slope = 0.2 * math.pi
   ...: offset = 0.3 * math.pi
   ...:
   ...: xs = np.arange(32, dtype=int)
   ...:
   ...: from_circuit = []
   ...: for x in range(32):
   ...:     qc = QuantumCircuit(control, target)
   ...:     qc.initialize(f"{x:05b}", control)
   ...:     qc.append(LinearPauliRotations(len(control), slope, offset), list(control) + list(target), [])
   ...:     from_circuit.append(Statevector(qc)[x].real)
   ...:
   ...: from_circuit = np.array(from_circuit)
   ...: expected = np.cos(slope * xs / 2 + offset / 2)
   ...:
   ...: np.set_printoptions(linewidth=80)
   ...: print(from_circuit)
   ...: print()
   ...: print(expected)
[ 0.89100652  0.70710678  0.4539905   0.15643447 -0.15643447 -0.4539905
 -0.70710678 -0.89100652 -0.98768834 -0.98768834 -0.89100652 -0.70710678
 -0.4539905  -0.15643447  0.15643447  0.4539905   0.70710678  0.89100652
  0.98768834  0.98768834  0.89100652  0.70710678  0.4539905   0.15643447
 -0.15643447 -0.4539905  -0.70710678 -0.89100652 -0.98768834 -0.98768834
 -0.89100652 -0.70710678]

[ 0.89100652  0.70710678  0.4539905   0.15643447 -0.15643447 -0.4539905
 -0.70710678 -0.89100652 -0.98768834 -0.98768834 -0.89100652 -0.70710678
 -0.4539905  -0.15643447  0.15643447  0.4539905   0.70710678  0.89100652
  0.98768834  0.98768834  0.89100652  0.70710678  0.4539905   0.15643447
 -0.15643447 -0.4539905  -0.70710678 -0.89100652 -0.98768834 -0.98768834
 -0.89100652 -0.70710678]
```

If we substituted $a/2 = \text{slope}$, $b/2 = \text{offset}$ into your first formula, we'd have the cosine as $\cos\bigl(\text{slope}\times x + \text{offset}\bigr)$, and it'd be the same if we replaced $a$ and $b$ with $\text{slope}$ and $\text{offset}$ in your second formula.

I think the correct fix is to keep the formula in our documentation the same, but fix the variable definitions to say $a = \text{slope} / 2$ and $b = \text{offset} / 2$.  Julien can also check my maths, though.

**Comment by Cryoris** [comment]
Yep I agree with the above 👍🏻 This comes orginally from https://arxiv.org/pdf/1806.06893.pdf, where in Fig. 2 you can see that it has the factor of 1/2. Fixing the variable definitions seems like the easiest way forward! 🙂

**Comment by jakelishman** [comment]
Thanks both!  Mattia - would you be interested in making the PR to fix things?

**Comment by Cryoris** [comment]
Fixed by #9891.
